src/proxy/http/profiles.rs

```diff
@@ -413,17 +504,23 @@ pub mod router {
         }
 
         fn call(&mut self, target: T) -> Self::Future {
-            let inner = self.inner.make(&target);
-            let stack = self.route_layer.clone().service(svc::shared(inner));
+            let mut make = IndexMap::with_capacity(1);
+            make.insert(target.clone(), self.inner.make(&target));
+
+            let concrete_router =
+                rt::Router::new_fixed(ConcreteDstRecognize::new(target.clone(), Vec::new()), make);
+
+            let stack = self
+                .route_layer
+                .clone()
+                .service(svc::shared(concrete_router.clone()));
             let default_route = target.clone().with_route(self.default_route.clone());
 
             let mut make = IndexMap::with_capacity(1);
             make.insert(default_route.clone(), stack.make(&default_route));
 
-            // Create a new fixed router; we can eagerly make the services
-            // and never expire the routes from the profile router cache
-            let router = Router::new_fixed(
-                Recognize {
+            let router = rt::Router::new_fixed(
```

Comment:
I know this method wasn't heavily commented to start, but keeping the existing comment about `new_fixed` and adding additional ones about why we need to make two routers now would be helpful.